pysph/base/particle_array.pyx

```diff
@@ -557,7 +557,7 @@ cdef class ParticleArray:
             if prop in self.stride.keys():
                 stride = self.stride[prop]
                 prop_len = len(particle_props[prop])
-                base_len = self.get_number_of_particles()
+                base_len = len(particle_props['h'])
```

Comment:
In general we cannot be sure that `h` exists as a prop in some cases, so perhaps pick any property with a unit stride and find its length.  Also, are we guaranteed that the particle props being passed are one dimensional i.e. `len(prop.shape) ==1`?